### ISSUE TITLE
[Feature Request]Support configuration of a "root-path" for entire project

### ISSUE THREAD

**Issue by andrelcm** [rio-labs/rio#129]
### Description

Currently, as far as I know, the default configuration of root path for web apps is / and that cannot be changed. So, for instance, we can start an app that answer requests on http://localhost:5000/ but not on http://localhost:5000/myapp/
If I am correct, the lack of this feature raises lots of problems with proxies, like nginx, when you run it in a subfolder. Although it is possible to configure nginx to strip the path before contacting the upstream, for some reason, that does not work with a rio app.

### Suggested Solution

I'd suggest two possible solutions (both could apply simultaneously): provide a "root path" configuration option and/or use only relative paths in entire project.

### Alternatives

I am not an expert in rio or related projects, but I tried setting somehow the root path option for uvicorn without any luck.

### Additional Context

I have a nginx with a subfolder "/myapp/" that redirects to the app running in "http://upstream:5000/". Without any further configuration, an access to http(s)://mynginx/myapp/ won't work, because it will call /myapp/ on upstream. Striping the path in nginx could work, and it almost worked on calling / , but the page loaded requests resources on /rio (in other words, from the root path), that does not exist on nginx. sub_filter does not solve the problem because it cannot catch some strings inside javascript. Mapping /rio to the upstream, despite beeing a poor solution, still does not work. However, even in this scenario, all resources are correctly loaded by the browser, but the main page complains that the address does not exist.

**Comment by mad-moo** [comment]
Hey!

To the best of my knowledge all paths in Rio are relative, so this is _supposed to work_. I'll try to reproduce this and get back to you. Thank you for the report!

**Comment by mad-moo** [comment]
I can reproduce this

**Comment by mad-moo** [comment]
4c667a2ba027f7acae4958ced2e499853a855c38 adds initial support for custom base URLs. This has required far more changes than anticipated and as such needs thorough testing before calling it finished.

@andrelcm if you run the development version please give it a shot and let me know how it goes. We'll also do more testing over the coming days.

The easiest way to set the base-url is using `rio run`:

```sh
python -m rio run --base-url "http://localhost/app"
```

The `as_fastapi` and `run_as_web_server` methods have also gotten a new parameter that works the same way.

**Comment by andrelcm** [comment]
Hi,

Thank you very much for the effort and so quick response.
I couple of questions, though.
I'm not an expert yet in python/rio, so... sorry if this is too obvious, but how can the development version? Should I get it directly from github?

Thanks in advance.

**Comment by mad-moo** [comment]
Yeah, since this isn't in any published release yet, you need to install directly from the source. Our [Contributing Guidelines](https://github.com/rio-labs/rio/blob/main/CONTRIBUTING.md#development-setup) have a section on how to build Rio locally. You can skip the forking part of course.

If you want to keep it simple you can also just wait for the next pre-release and then install from pypi. We'll ship this code soon (albeit marked as experimental) so we can get real-world feedback.

**Comment by mad-moo** [comment]
Hey @andrelcm !

A new pre-release is out, and with that experimental support for the base URL. You can easily get it from PyPI:

```sh
pip install --upgrade --pre rio-ui
```

(Note the `--pre`. Without that flag `pip` ignores pre-releases like this one)

**Comment by mad-moo** [comment]
This has been in the codebase for a while now, but since nobody is actively testing it, I'll keep the issue open
